### PR TITLE
add prune feature

### PR DESCRIPTION
**Pull request by tdantas**
ability to prune ablauf ast.

basically the idea is to remove empty :ast/nodes from the tree.  
for instance
```
#:ast{:type :ast/seq
      :nodes [#:ast{:type :ast/leaf :action :action/log :payload :b}
              #:ast{:type :ast/seq :nodes []}]}]
```

after prune will be something like
```
#:ast{:type :ast/seq
      :nodes [#:ast{:type :ast/leaf :action :action/log :payload :b}]}]
```

### CODE REVIEW

**Comment by wdullaer** [comment]
Can this maybe be implemented in the ablauf library?
I have implemented similar functionality in the root-api-ui code.

File: src/ablauf/job.clj [hunk 1]
```diff
@@ -219,3 +219,19 @@
     (eligible? ast) :job/pending
     :else           (throw (ex-info "Wrong AST job state"
                                     {}))))
+
+(defn prune
+  "Remove empties :ast/nodes leaves"
+  [ast]
+  (loop [zipper (zip/next (ast-zip ast))]
+    (if (zip/end? zipper)
+      (zip/node zipper)
```
Comment (pyr) [inline src/ablauf/job.clj:228]: shouldn't this be `zip/root`?

File: src/ablauf/job.clj [hunk 2]
```diff
@@ -219,3 +219,19 @@
...
+      (let [{:ast/keys [nodes]} (zip/node zipper)]
+        (cond
+          (and nodes
```
Comment (pyr) [inline src/ablauf/job.clj:231]: make it more explicit
```suggestion
          (and (some? nodes)
```

File: test/ablauf/job_prune.clj [hunk 3]
```diff
@@ -0,0 +1,37 @@
+(ns ablauf.job-prune
+  (:require [ablauf.job :refer [prune]]
+            [ablauf.job.ast :as ast]
+            [clojure.test :refer [deftest is testing]]))
+
```
Comment (pyr) [inline test/ablauf/job_prune.clj:5]: I think you need a test which tests a nested empty seq

Comment (tdantas) [reply test/ablauf/job_prune.clj:5]: updated f42bb46

**Review by pyr — APPROVED** [review]
